date-parser

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+
+# Used to check which machine we are running on
+unameOut="$(uname -s)"
+MAC_OS="Darwin"
+LINUX_OS="Linux"
+
+defaultDateFormat="%Y-%m-%d"
+
+parseInputDate() {
+    dateToParse="${1}"
+    # Validate input date and format it
+    # Take into account different implementations of `date` commant
+    if [ "${unameOut}" = "${LINUX_OS}" ]
+    then
+        requiredDate="${dateToParse:-today}"
+        formattedDate="$( date -d "${requiredDate}" +${defaultDateFormat})"
+    elif [ "${unameOut}" = "${MAC_OS}" ]
+    then
+        # On BSD based systems (i.e. Mac), date cannot accept any input as a string
+        # So we have to enforce a specific format to validate against
+        todaysDateMacOs="$(date -j +${defaultDateFormat})"
+        requiredDate="${dateToParse:-${todaysDateMacOs}}"
+
+        validationResults="$(date -j -f "${defaultDateFormat}" "${requiredDate}" > /dev/null 2>&1)"
+
+        if [ "$?" = "1" ]
+        then
+            echo "Invalid date format ! Received: \"${requiredDate}\". Date should match: ${defaultDateFormat}"
+            exit 1
+        fi
+        # A bit more context on -f -> "-f input_fmt new_date +output_fmt"
+        # @see `man date` on BSD/Mac OS for more details
+        formattedDate="$(date -j -f "${defaultDateFormat}" ${requiredDate} +${defaultDateFormat})"
```

Comment:
Interesting, I wasn't aware variable were global by default (unless to prefix its initialization with `local`).

I think its best to `echo` the variable and then `return 0` rather than using global variables.
```bash
  # A bit more context on -f -> "-f input_fmt new_date +output_fmt"
  # @see `man date` on BSD/Mac OS for more details 
  echo "$(date -j -f "${defaultDateFormat}" ${requiredDate} +${defaultDateFormat})"
  return 0
fi

echo "Unsupported platform '${unameOut}'"
return 1
```